src/parquet/file/reader-internal.cc

```diff
@@ -293,10 +255,13 @@ void SerializedFile::ParseMetaData() {
   if (bytes_read != metadata_len) {
     throw ParquetException("Invalid parquet file. Could not read metadata bytes.");
   }
-  DeserializeThriftMsg(&metadata_buffer[0], &metadata_len, &metadata_);
 
-  schema::FlatSchemaConverter converter(&metadata_.schema[0], metadata_.schema.size());
-  schema_.Init(converter.Convert());
+  auto metadata_ptr = std::unique_ptr<format::FileMetaData>(new format::FileMetaData());
+  metadata_ = metadata_ptr.get();
```

Comment:
Ideally we should not retain this pointer at this level. Can you create a JIRA about routing all metadata access through the new classes?